### PR TITLE
add a MarshalJSON method

### LINKED ISSUE #12: JSON printout of the combined claims-set
We should provide an interface to print the combined platform and realm claims-sets in JSON format.

### PR DESCRIPTION
**Pull request by thomas-fossati**
Fix #12

Signed-off-by: Thomas Fossati <thomas.fossati@arm.com>

### CODE REVIEW

File: evidence.go [hunk 1]
```diff
@@ -27,6 +27,30 @@ type Evidence struct {
 	collection     *Collection
 }
 
+func (e *Evidence) MarshalJSON() ([]byte, error) {
```
Comment (yogeshbdeshpande) [inline evidence.go:37]: Please add UnMarshaler as well.

Comment (thomas-fossati) [reply evidence.go:37]: done in 772e0cf

File: evidence.go [hunk 2]
```diff
@@ -5,26 +5,85 @@ import (
...
+	if err != nil {
+		return nil, fmt.Errorf("error serializing platform claims: %w", err)
+	}
+
+	rj, err := e.realmClaims.ToJSON()
+	if err != nil {
+		return nil, fmt.Errorf("error serializing realm claims: %w", err)
+	}
+
+	ej := "{" +
```
Comment (yogeshbdeshpande) [inline evidence.go]: Question: Why do we need to do this: Should we not use the JSON Collection struct to set the Individual JSON containers and then Marshal that struct to get ej?

Comment (thomas-fossati) [reply evidence.go]: absolutely.  The JSON collection was introduced in [a4fb8bc](https://github.com/veraison/ccatoken/pull/13/commits/a4fb8bc3745ddaba4e3a7d9e6e6b7b2eeca3d0ae) for the unmarshaler and I forgot to retrofit it into the marshaler.

Comment (thomas-fossati) [reply evidence.go]: done in 1ff60d0

**Review by yogeshbdeshpande — APPROVED** [review]
Thanks for addressing my comments: LGTM!

**Pull request merged**